osquery/main/ctl.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <iostream>
+
+#include <string.h>
+
+#include <glog/logging.h>
+
+#include <osquery/core.h>
+
+namespace osquery {
+
+void printCtlHelp() {
+  std::cout << " osqueryctl\n";
+}
+
+void printCtlHelpAndExit() {
+  printCtlHelp();
+  exit(0);
+}
+
+void doSomething() {
+  std::cout << "doing something\n";
+}
+
+typedef void (*PositionalArgFunction)(void);
+
+typedef std::map<std::string, PositionalArgFunction> CommandMap;
+
+const CommandMap kCtlArguments = CommandMap{
+  {"start", doSomething},
+  {"stop", doSomething},
+  {"config-check", doSomething},
+  {"status", doSomething},
+  {"info", doSomething},
+};
+
+}
+
+int main(int argc, char* argv[]) {
+  osquery::initOsquery(argc, argv);
+
+  // if osqueryctl was ran on it's own or with a positional argument "help"
+  if (argc < 2 || strncmp(argv[1], "help", 4) == 0) {
```

Comment:
You might not be able to evaluate help since `initOsquery` tries to handle help in argv. There are definitions for each osquery tool (right now only daemon and shell) and they make help/usage decisions.